google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/instance/AppEngineServerModel.java

```diff
@@ -47,24 +49,33 @@
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
 
 /**
  * @author nik
  */
-public class AppEngineServerModel implements ServerModel, DeploysArtifactsOnStartupOnly {
+public class AppEngineServerModel implements ServerModel, DeploysArtifactsOnStartupOnly,
+    RunConfiguration, Cloneable {
 
-  private ArtifactPointer myArtifactPointer;
-  private int myPort = 8080;
-  private String myServerParameters = "";
-  private CommonModel myCommonModel;
+  private ArtifactPointer artifactPointer;
+  private CommonModel commonModel;
+
+  private AppEngineModelSettings settings = new AppEngineModelSettings();
 
   @Override
   public J2EEServerInstance createServerInstance() throws ExecutionException {
-    return new AppEngineServerInstance(myCommonModel);
+    return new AppEngineServerInstance(commonModel);
   }
 
+  // TODO(joaomartins): Consider adding something here so we get a nice log pane
```

Comment:
I was thinking of working on these immediately after I push this PR. (i.e., these TODOs would effectively be my issue list). But I can create an issue for this one, sure.